aii-pxelinux/src/test/perl/write_grub2_config.t

```diff
@@ -0,0 +1,70 @@
+use strict;
+use warnings;
+use Test::More;
+use Test::Quattor qw(pxelinux_base_config);
+use NCM::Component::PXELINUX::constants qw(:pxe_variants :pxe_constants);
+use NCM::Component::pxelinux;
+use CAF::FileWriter;
+use CAF::Object;
+use Sys::Hostname;
+use Readonly;
+
+
+=pod
+
+=head1 SYNOPSIS
+
+Tests for the C<write_grub2_config> method.
+
+=cut
+
+Readonly my $TEST_EFI_LINUX_CMD => GRUB2_EFI_LINUX_CMD_DEFAULT;
+Readonly my $TEST_EFI_INITRD_CMD => 'initrdefi';
+Readonly my $NBPDIR_GRUB2_VALUE => '/grub/config_files';
+Readonly my $GRUB2_EFI_KERNEL_ROOT_VALUE => '/quat/or';
+
+$CAF::Object::NoAction = 1;
+
+our $this_app = $main::this_app;
+
+
+sub check_config {
+    my ($comp, $cfg, $kernel_root, $test_msg) = @_;
+
+    $comp->write_grub2_config($cfg);
+
+    # Retrieve config file name matching the configuration
+    my $fp = NCM::Component::pxelinux::filepath($cfg, PXE_VARIANT_GRUB2);
+    
+    # Check config file contents
+    my $fh = get_file($fp);
+    my $hostname = hostname();
+    like($fh, qr{^set default=0$}m, "default kernel ($test_msg)");
+    like($fh, qr{^set timeout=\d+$}m, "Grub2 menu timeout ($test_msg)");
+    like($fh, qr(^menuentry\s"Install\s[\w\-\s\(\)\[\]]+"\s{$)m, "Grub2 menu entry ($test_msg)");
```

Comment:
escape the `{` to fix a warning in recent perl versions (and prevent future tests to fail with `Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/^menuentry\s"Install\s[\w\-\s\(\)\[\]]+"\s{ <-- HERE $/ at src/test/perl/write_grub2_config.t line 44.`)